git-switch.rb

```diff
@@ -34,7 +33,12 @@
 when :modified
   branches = `git for-each-ref --format="%(refname:short)" --sort='-authordate' refs/heads --count #{options[:count]}`.split
 else
-  branches = `git reflog | grep "checkout: moving" | cut -d' ' -f8`.split.uniq.drop(1).take(options[:count])
+  branches = `git reflog | grep "checkout: moving" | cut -d' ' -f8`.split.uniq.drop(1)
+
+  # Remove deleted branches from the result set
+  branches &= `git branch`.split unless options[:show_deleted_branches]
```

Comment:
Great catch! 

I was experimenting with an option to show deleted branches but I agree with you that it doesn't make sense.